src/test/java/com/marklogic/kafka/connect/sink/WriteFromKafkaTest.java

```diff
@@ -0,0 +1,100 @@
+    private final String TOPIC = "test-topic";
+    private final String KEY = String.format("key-%s", UUID.randomUUID());
+
+    private EmbeddedKafkaCluster kafka;
+
+    @BeforeEach
+    void setupKafka() {
+        provisionKafkaWithConnectAndMarkLogicConnector();
+        kafka.start();
+    }
+
+    @AfterEach
+    void tearDownKafka() {
+        kafka.stop();
+    }
+
+    @Test
+    void shouldWaitForKeyedRecordsToBePublished() throws InterruptedException {
+        Integer NUM_RECORDS = 2;
+        sendSomeJsonMessages(NUM_RECORDS);
+
+        // The processing is fast, but without the short pause, the count comes up short from time to time.
+        // If this test fails intermittently, this might be the reason.
+        Thread.sleep(500);
+        assertMarkLogicDocumentsExistInCollection(ML_COLLECTION, NUM_RECORDS,
+            format("Expected to find %d records in the ML database", NUM_RECORDS));
+    }
+
+    private void provisionKafkaWithConnectAndMarkLogicConnector() {
+        kafka = provisionWith(
+            newClusterConfig()
+                .configure(
+                    kafkaConnect()
+                        .deployConnector(connectorConfig(TOPIC, KEY))
+                        .with(KafkaConfig$.MODULE$.NumPartitionsProp(), "5")
+                )
+        );
+    }
+
+    private Properties connectorConfig(final String topic, final String key) {
+        return MarkLogicSinkConnectorConfigBuilder.create()
+            .withTopic(topic)
+            .withKey(key)
+            .with(MarkLogicSinkConfig.CONNECTION_HOST, testConfig.getHost())
+            .with(MarkLogicSinkConfig.CONNECTION_PORT, testConfig.getRestPort())
+            .with(MarkLogicSinkConfig.CONNECTION_USERNAME, testConfig.getUsername())
+            .with(MarkLogicSinkConfig.CONNECTION_PASSWORD, testConfig.getPassword())
+            .with(MarkLogicSinkConfig.DOCUMENT_COLLECTIONS, ML_COLLECTION)
+            .with(MarkLogicSinkConfig.DMSDK_BATCH_SIZE, 1)
+            .build();
+    }
+
+    private void sendSomeJsonMessages(Integer numberOfRecords) throws InterruptedException {
+        List<KeyValue<String, String>> records = new ArrayList<>();
+        for (int i = 0; i < numberOfRecords; i++) {
+            records.add(new KeyValue<>("aggregate", "{\"A\": \"" + i + "\"}"));
+        }
+        kafka.send(SendKeyValues.to(TOPIC, records));
+    }
+
+    private void assertMarkLogicDocumentsExistInCollection(String collection, Integer numRecords, String message) {
+        StructuredQueryBuilder qb = new StructuredQueryBuilder();
+        StructuredQueryDefinition queryDefinition = qb.collection(collection);
+        SearchHandle results = getDatabaseClient().newQueryManager().search(queryDefinition, new SearchHandle());
+        assertEquals(numRecords.longValue(), results.getTotalResults(), message);
+    }
+
+    private void assertRecordsOnTheTopic(String topic, Integer num_records, Integer timeCt) throws InterruptedException {
```

Comment:
Can remove this, as it's unused (though might be worth pasting into the JIRA ticket for future reference if needed in the future).